src/sentry/auth/helper.py

```diff
@@ -671,8 +699,14 @@ def finish_pipeline(self):
 
         return response
 
+    @property
+    def auth_handler(self):
```

Comment:
wouldn't you want this to not create a new object on every access? e.g. should this be memoized?

Reply:
I believe it would create it only once in the lifespan of the typical `AuthHelper` object, but sure, I can make it to an instance variable.

Reply:
property just removes the need to put `()` on a function call, so everything time you do `self.auth_handler` that is the equiv of doing `self.auth_handler()` without the `@property` decorator. If you want it to lazy-instantiate, you can use `@cached_decorator` or `@memoize` (I forget which one we have)

Reply:
Understood. There's no need to lazy-instantiate. I was just extracting duplicated code and I think my Java instincts caused me to gravitate toward a method rather than an attribute. Either way, there are no performance implications -- `AuthIdentityHandler` is cheap to construct.

Reply:
Never mind, see https://github.com/getsentry/sentry/pull/26571/commits/3b3ae950ddf958b4d5da5c5380c453c26381709f. It does need to be instantiated upon use because `AuthHelper.auth_provider` is mutable.